### DISCUSSION TITLE
PBP preconditioners

### DISCUSSION THREAD

**Discussion by nanguaxiaofendui** [idaholab/moose#23688] [Q&A General] [answered]
Hello all,
May I ask whether I can print out the PBP preprocessing matrix M？How to output?

**Comment by GiudGiud** [comment, marked as answer]
-ksp_view_pmat
On the command line or in the petsc_options parameter

**Reply by nanguaxiaofendui** [reply]
I have tried, other preprocessing can be output with this command, but PBP can not, is there any other way? thank you

**Reply by GiudGiud** [reply]
You ll have to check the petsc manual to get more output flags. 
there s also -ksp_view_mat
